File: decompiled_melonDS/sources/me/magnum/melonds/ui/settings/fragments/SystemPreferencesFragment.java
package me.magnum.melonds.ui.settings.fragments;

import android.net.Uri;
import android.os.Build;
import android.os.Bundle;
import androidx.fragment.app.b1;
import androidx.lifecycle.s;
import androidx.lifecycle.s0;
import androidx.preference.Preference;
import androidx.preference.SwitchPreference;
import androidx.preference.p;
import kf.x1;
import me.magnum.melonds.R;
import me.magnum.melonds.common.camera.DSiCameraSource;
import zc.g0;
import zc.x;
/* compiled from: r8-map-id-68a076d6c8e19c01d0388d4729d627070c4f7cff1f6777b7d5e4497c81f45b39 */
/* loaded from: classes.dex */
public final class SystemPreferencesFragment extends Hilt_SystemPreferencesFragment implements ci.l {
    public x1 Y;
    public final h.c Z;

    /* renamed from: b0  reason: collision with root package name */
    public final h.c f9605b0;

    /* renamed from: c0  reason: collision with root package name */
    public final h.c f9606c0;

    /* renamed from: d0  reason: collision with root package name */
    public final h.c f9607d0;

    public SystemPreferencesFragment() {
        h.c registerForActivityResult = registerForActivityResult(new b1(1), new h.b(this) { // from class: me.magnum.melonds.ui.settings.fragments.l
            public final /* synthetic */ SystemPreferencesFragment B;

            {
                this.B = this;
            }

            @Override // h.b
            public final void a(Object obj) {
                int i2 = r2;
                SystemPreferencesFragment systemPreferencesFragment = this.B;
                Uri uri = (Uri) obj;
                switch (i2) {
                    case 0:
                        if (uri != null) {
                            s f8 = s0.f(systemPreferencesFragment);
                            gd.f fVar = g0.f15015a;
                            x.v(f8, gd.e.L, null, new SystemPreferencesFragment$backupInternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                    case DSiCameraSource.FrontCamera /* 1 */:
                        if (uri != null) {
                            s f10 = s0.f(systemPreferencesFragment);
                            gd.f fVar2 = g0.f15015a;
                            x.v(f10, gd.e.L, null, new SystemPreferencesFragment$backupExternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                    case 2:
                        if (uri != null) {
                            s f11 = s0.f(systemPreferencesFragment);
                            gd.f fVar3 = g0.f15015a;
                            x.v(f11, gd.e.L, null, new SystemPreferencesFragment$restoreInternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                    default:
                        if (uri != null) {
                            s f12 = s0.f(systemPreferencesFragment);
                            gd.f fVar4 = g0.f15015a;
                            x.v(f12, gd.e.L, null, new SystemPreferencesFragment$restoreExternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                }
            }
        });
        registerForActivityResult.getClass();
        this.Z = registerForActivityResult;
        h.c registerForActivityResult2 = registerForActivityResult(new b1(1), new h.b(this) { // from class: me.magnum.melonds.ui.settings.fragments.l
            public final /* synthetic */ SystemPreferencesFragment B;

            {
                this.B = this;
            }

            @Override // h.b
            public final void a(Object obj) {
                int i2 = r2;
                SystemPreferencesFragment systemPreferencesFragment = this.B;
                Uri uri = (Uri) obj;
                switch (i2) {
                    case 0:
                        if (uri != null) {
                            s f8 = s0.f(systemPreferencesFragment);
                            gd.f fVar = g0.f15015a;
                            x.v(f8, gd.e.L, null, new SystemPreferencesFragment$backupInternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                    case DSiCameraSource.FrontCamera /* 1 */:
                        if (uri != null) {
                            s f10 = s0.f(systemPreferencesFragment);
                            gd.f fVar2 = g0.f15015a;
                            x.v(f10, gd.e.L, null, new SystemPreferencesFragment$backupExternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                    case 2:
                        if (uri != null) {
                            s f11 = s0.f(systemPreferencesFragment);
                            gd.f fVar3 = g0.f15015a;
                            x.v(f11, gd.e.L, null, new SystemPreferencesFragment$restoreInternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                    default:
                        if (uri != null) {
                            s f12 = s0.f(systemPreferencesFragment);
                            gd.f fVar4 = g0.f15015a;
                            x.v(f12, gd.e.L, null, new SystemPreferencesFragment$restoreExternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                }
            }
        });
        registerForActivityResult2.getClass();
        this.f9605b0 = registerForActivityResult2;
        h.c registerForActivityResult3 = registerForActivityResult(new b1(1), new h.b(this) { // from class: me.magnum.melonds.ui.settings.fragments.l
            public final /* synthetic */ SystemPreferencesFragment B;

            {
                this.B = this;
            }

            @Override // h.b
            public final void a(Object obj) {
                int i2 = r2;
                SystemPreferencesFragment systemPreferencesFragment = this.B;
                Uri uri = (Uri) obj;
                switch (i2) {
                    case 0:
                        if (uri != null) {
                            s f8 = s0.f(systemPreferencesFragment);
                            gd.f fVar = g0.f15015a;
                            x.v(f8, gd.e.L, null, new SystemPreferencesFragment$backupInternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                    case DSiCameraSource.FrontCamera /* 1 */:
                        if (uri != null) {
                            s f10 = s0.f(systemPreferencesFragment);
                            gd.f fVar2 = g0.f15015a;
                            x.v(f10, gd.e.L, null, new SystemPreferencesFragment$backupExternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                    case 2:
                        if (uri != null) {
                            s f11 = s0.f(systemPreferencesFragment);
                            gd.f fVar3 = g0.f15015a;
                            x.v(f11, gd.e.L, null, new SystemPreferencesFragment$restoreInternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                    default:
                        if (uri != null) {
                            s f12 = s0.f(systemPreferencesFragment);
                            gd.f fVar4 = g0.f15015a;
                            x.v(f12, gd.e.L, null, new SystemPreferencesFragment$restoreExternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                }
            }
        });
        registerForActivityResult3.getClass();
        this.f9606c0 = registerForActivityResult3;
        h.c registerForActivityResult4 = registerForActivityResult(new b1(1), new h.b(this) { // from class: me.magnum.melonds.ui.settings.fragments.l
            public final /* synthetic */ SystemPreferencesFragment B;

            {
                this.B = this;
            }

            @Override // h.b
            public final void a(Object obj) {
                int i2 = r2;
                SystemPreferencesFragment systemPreferencesFragment = this.B;
                Uri uri = (Uri) obj;
                switch (i2) {
                    case 0:
                        if (uri != null) {
                            s f8 = s0.f(systemPreferencesFragment);
                            gd.f fVar = g0.f15015a;
                            x.v(f8, gd.e.L, null, new SystemPreferencesFragment$backupInternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                    case DSiCameraSource.FrontCamera /* 1 */:
                        if (uri != null) {
                            s f10 = s0.f(systemPreferencesFragment);
                            gd.f fVar2 = g0.f15015a;
                            x.v(f10, gd.e.L, null, new SystemPreferencesFragment$backupExternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                    case 2:
                        if (uri != null) {
                            s f11 = s0.f(systemPreferencesFragment);
                            gd.f fVar3 = g0.f15015a;
                            x.v(f11, gd.e.L, null, new SystemPreferencesFragment$restoreInternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                    default:
                        if (uri != null) {
                            s f12 = s0.f(systemPreferencesFragment);
                            gd.f fVar4 = g0.f15015a;
                            x.v(f12, gd.e.L, null, new SystemPreferencesFragment$restoreExternalLayoutLauncher$1$1(systemPreferencesFragment, uri, null), 2);
                            return;
                        }
                        return;
                }
            }
        });
        registerForActivityResult4.getClass();
        this.f9607d0 = registerForActivityResult4;
    }

    @Override // ci.l
    public final String getTitle() {
        String string = getString(R.string.category_system);
        string.getClass();
        return string;
    }

    @Override // androidx.preference.a0
    public final void onCreatePreferences(Bundle bundle, String str) {
        setPreferencesFromResource(R.xml.pref_system, str);
        Preference findPreference = findPreference("enable_jit");
        findPreference.getClass();
        SwitchPreference switchPreference = (SwitchPreference) findPreference;
        String[] strArr = Build.SUPPORTED_64_BIT_ABIS;
        strArr.getClass();
        if (strArr.length == 0) {
            switchPreference.setEnabled(false);
            switchPreference.e(false);
            switchPreference.setSummary(R.string.jit_not_supported);
        }
        Preference findPreference2 = findPreference("backup_internal_layout");
        if (findPreference2 != null) {
            findPreference2.setOnPreferenceClickListener(new p(this) { // from class: me.magnum.melonds.ui.settings.fragments.m
                public final /* synthetic */ SystemPreferencesFragment B;

                {
                    this.B = this;
                }

                @Override // androidx.preference.p
                public final void a(Preference preference) {
                    switch (r2) {
                        case 0:
                            this.B.Z.a(null);
                            return;
                        case DSiCameraSource.FrontCamera /* 1 */:
                            this.B.f9605b0.a(null);
                            return;
                        case 2:
                            this.B.f9606c0.a(null);
                            return;
                        default:
                            this.B.f9607d0.a(null);
                            return;
                    }
                }
            });
        }
        Preference findPreference3 = findPreference("backup_external_layout");
        if (findPreference3 != null) {
            findPreference3.setOnPreferenceClickListener(new p(this) { // from class: me.magnum.melonds.ui.settings.fragments.m
                public final /* synthetic */ SystemPreferencesFragment B;

                {
                    this.B = this;
                }

                @Override // androidx.preference.p
                public final void a(Preference preference) {
                    switch (r2) {
                        case 0:
                            this.B.Z.a(null);
                            return;
                        case DSiCameraSource.FrontCamera /* 1 */:
                            this.B.f9605b0.a(null);
                            return;
                        case 2:
                            this.B.f9606c0.a(null);
                            return;
                        default:
                            this.B.f9607d0.a(null);
                            return;
                    }
                }
            });
        }
        Preference findPreference4 = findPreference("restore_internal_layout");
        if (findPreference4 != null) {
            findPreference4.setOnPreferenceClickListener(new p(this) { // from class: me.magnum.melonds.ui.settings.fragments.m
                public final /* synthetic */ SystemPreferencesFragment B;

                {
                    this.B = this;
                }

                @Override // androidx.preference.p
                public final void a(Preference preference) {
                    switch (r2) {
                        case 0:
                            this.B.Z.a(null);
                            return;
                        case DSiCameraSource.FrontCamera /* 1 */:
                            this.B.f9605b0.a(null);
                            return;
                        case 2:
                            this.B.f9606c0.a(null);
                            return;
                        default:
                            this.B.f9607d0.a(null);
                            return;
                    }
                }
            });
        }
        Preference findPreference5 = findPreference("restore_external_layout");
        if (findPreference5 != null) {
            findPreference5.setOnPreferenceClickListener(new p(this) { // from class: me.magnum.melonds.ui.settings.fragments.m
                public final /* synthetic */ SystemPreferencesFragment B;

                {
                    this.B = this;
                }

                @Override // androidx.preference.p
                public final void a(Preference preference) {
                    switch (r2) {
                        case 0:
                            this.B.Z.a(null);
                            return;
                        case DSiCameraSource.FrontCamera /* 1 */:
                            this.B.f9605b0.a(null);
                            return;
                        case 2:
                            this.B.f9606c0.a(null);
                            return;
                        default:
                            this.B.f9607d0.a(null);
                            return;
                    }
                }
            });
        }
    }
}

File: app/src/main/java/me/magnum/melonds/ui/settings/fragments/SystemPreferencesFragment.kt
package me.magnum.melonds.ui.settings.fragments

import android.os.Build
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import androidx.activity.result.contract.ActivityResultContracts
import androidx.lifecycle.lifecycleScope
import androidx.preference.ListPreference
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import androidx.preference.SwitchPreference
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import me.magnum.melonds.R
import me.magnum.melonds.common.DirectoryAccessValidator
import me.magnum.melonds.common.UriPermissionManager
import me.magnum.melonds.impl.SettingsBackupManager
import me.magnum.melonds.ui.settings.PreferenceFragmentHelper
import me.magnum.melonds.ui.settings.PreferenceFragmentTitleProvider
import me.magnum.melonds.ui.settings.preferences.StoragePickerPreference
import javax.inject.Inject

@AndroidEntryPoint
class SystemPreferencesFragment : BasePreferenceFragment(), PreferenceFragmentTitleProvider {

    @Inject lateinit var uriPermissionManager: UriPermissionManager
    @Inject lateinit var directoryAccessValidator: DirectoryAccessValidator
    @Inject lateinit var settingsBackupManager: SettingsBackupManager
    private val helper by lazy { PreferenceFragmentHelper(this, uriPermissionManager, directoryAccessValidator) }
    private var updatingMirrorPreference = false

    private val backupInternalLayoutLauncher = registerForActivityResult(ActivityResultContracts.OpenDocumentTree()) { uri ->
        if (uri != null) {
            lifecycleScope.launch(Dispatchers.IO) {
                runCatching { settingsBackupManager.backupInternalLayout(uri) }
                    .onSuccess {
                        withContext(Dispatchers.Main) {
                            AlertDialog.Builder(requireContext())
                                .setMessage(R.string.internal_layout_backup_success)
                                .setPositiveButton(android.R.string.ok, null)
                                .show()
                        }
                    }
                    .onFailure {
                        withContext(Dispatchers.Main) {
                            Toast.makeText(requireContext(), R.string.internal_layout_backup_error, Toast.LENGTH_SHORT).show()
                        }
                    }
            }
        }
    }

    private val backupExternalLayoutLauncher = registerForActivityResult(ActivityResultContracts.OpenDocumentTree()) { uri ->
        if (uri != null) {
            lifecycleScope.launch(Dispatchers.IO) {
                runCatching { settingsBackupManager.backupExternalLayout(uri) }
                    .onSuccess {
                        withContext(Dispatchers.Main) {
                            AlertDialog.Builder(requireContext())
                                .setMessage(R.string.external_layout_backup_success)
                                .setPositiveButton(android.R.string.ok, null)
                                .show()
                        }
                    }
                    .onFailure {
                        withContext(Dispatchers.Main) {
                            Toast.makeText(requireContext(), R.string.external_layout_backup_error, Toast.LENGTH_SHORT).show()
                        }
                    }
            }
        }
    }

    private val restoreInternalLayoutLauncher = registerForActivityResult(ActivityResultContracts.OpenDocumentTree()) { uri ->
        if (uri != null) {
            lifecycleScope.launch(Dispatchers.IO) {
                runCatching { settingsBackupManager.restoreInternalLayout(uri) }
                    .onSuccess {
                        withContext(Dispatchers.Main) {
                            AlertDialog.Builder(requireContext())
                                .setMessage(R.string.internal_layout_restore_success)
                                .setPositiveButton(android.R.string.ok, null)
                                .show()
                        }
                    }
                    .onFailure {
                        withContext(Dispatchers.Main) {
                            Toast.makeText(requireContext(), R.string.internal_layout_restore_error, Toast.LENGTH_SHORT).show()
                        }
                    }
            }
        }
    }

    private val restoreExternalLayoutLauncher = registerForActivityResult(ActivityResultContracts.OpenDocumentTree()) { uri ->
        if (uri != null) {
            lifecycleScope.launch(Dispatchers.IO) {
                runCatching { settingsBackupManager.restoreExternalLayout(uri) }
                    .onSuccess {
                        withContext(Dispatchers.Main) {
                            AlertDialog.Builder(requireContext())
                                .setMessage(R.string.external_layout_restore_success)
                                .setPositiveButton(android.R.string.ok, null)
                                .show()
                        }
                    }
                    .onFailure {
                        withContext(Dispatchers.Main) {
                            Toast.makeText(requireContext(), R.string.external_layout_restore_error, Toast.LENGTH_SHORT).show()
                        }
                    }
            }
        }
    }

    override fun getTitle() = getString(R.string.category_system)

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.pref_system, rootKey)
        val jitPreference = findPreference<SwitchPreference>("enable_jit")!!
        val stormCompilerPreference = findPreference<SwitchPreference>("enable_storm_compiler")!!
        val mirrorPreference = findPreference<SwitchPreference>("save_internal_config_as_file")!!

        val appLanguagePreference = findPreference<ListPreference>("app_language")
        appLanguagePreference?.setOnPreferenceChangeListener { _, newValue ->
            val langTag = newValue as? String ?: "default"
            if (langTag == "default") {
                androidx.appcompat.app.AppCompatDelegate.setApplicationLocales(androidx.core.os.LocaleListCompat.getEmptyLocaleList())
            } else {
                androidx.appcompat.app.AppCompatDelegate.setApplicationLocales(androidx.core.os.LocaleListCompat.forLanguageTags(langTag))
            }
            true
        }

        if (Build.SUPPORTED_64_BIT_ABIS.isEmpty()) {
            jitPreference.isChecked = false
            jitPreference.isVisible = false
            stormCompilerPreference.isChecked = false
            stormCompilerPreference.isVisible = false
        } else {
            // Ensure only one compiler is enabled on launch (default: STORM-Compiler)
            if (stormCompilerPreference.isChecked && jitPreference.isChecked) {
                jitPreference.isChecked = false
                preferenceManager.sharedPreferences?.edit()?.putBoolean("enable_jit", false)?.apply()
            } else if (!stormCompilerPreference.isChecked && !jitPreference.isChecked) {
                stormCompilerPreference.isChecked = true
                preferenceManager.sharedPreferences?.edit()?.putBoolean("enable_storm_compiler", true)?.apply()
            }

            jitPreference.setOnPreferenceChangeListener { _, newValue ->
                if (newValue == true) {
                    stormCompilerPreference.isChecked = false
                }
                true
            }
            stormCompilerPreference.setOnPreferenceChangeListener { _, newValue ->
                if (newValue == true) {
                    jitPreference.isChecked = false
                }
                true
            }
        }

        mirrorPreference.setOnPreferenceChangeListener { _, newValue ->
            if (updatingMirrorPreference) {
                return@setOnPreferenceChangeListener true
            }
            if (newValue != true) {
                preferenceManager.sharedPreferences?.edit()?.putBoolean("save_internal_config_as_file", false)?.apply()
                return@setOnPreferenceChangeListener true
            }

            lifecycleScope.launch(Dispatchers.IO) {
                try {
                    val mirrorDirectory = settingsBackupManager.getActiveMirrorDirectory()
                    val hasMirror = mirrorDirectory != null && runCatching { settingsBackupManager.hasMirrorAt(mirrorDirectory) }.getOrDefault(false)

                    withContext(Dispatchers.Main) {
                        if (!isAdded) return@withContext
                        val ctx = context ?: return@withContext

                        if (!hasMirror || mirrorDirectory == null) {
                            setMirrorEnabled(mirrorPreference)
                            settingsBackupManager.requestMirrorWrite()
                        } else {
                            AlertDialog.Builder(ctx)
                                .setTitle(R.string.settings_mirror_detected_title)
                                .setMessage(R.string.settings_mirror_detected_message)
                                .setPositiveButton(R.string.settings_mirror_restore) { _, _ ->
                                    setMirrorEnabled(mirrorPreference)
                                    lifecycleScope.launch(Dispatchers.IO) {
                                        runCatching {
                                            settingsBackupManager.restoreMirrorFrom(mirrorDirectory)
                                            settingsBackupManager.requestMirrorWrite()
                                        }
                                    }
                                }
                                .setNegativeButton(R.string.settings_mirror_ignore) { _, _ ->
                                    setMirrorEnabled(mirrorPreference)
                                    lifecycleScope.launch(Dispatchers.IO) {
                                        runCatching {
                                            settingsBackupManager.overwriteMirrorAt(mirrorDirectory)
                                            settingsBackupManager.requestMirrorWrite()
                                        }
                                    }
                                }
                                .setOnCancelListener {
                                    updatingMirrorPreference = true
                                    mirrorPreference.isChecked = false
                                    preferenceManager.sharedPreferences?.edit()?.putBoolean("save_internal_config_as_file", false)?.apply()
                                    updatingMirrorPreference = false
                                }
                                .show()
                        }
                    }
                } catch (e: Exception) {
                    withContext(Dispatchers.Main) {
                        if (isAdded) {
                            setMirrorEnabled(mirrorPreference)
                            settingsBackupManager.requestMirrorWrite()
                        }
                    }
                }
            }
            false
        }

        findPreference<Preference>("backup_internal_layout")?.setOnPreferenceClickListener {
            backupInternalLayoutLauncher.launch(null)
            true
        }
        findPreference<Preference>("backup_external_layout")?.setOnPreferenceClickListener {
            backupExternalLayoutLauncher.launch(null)
            true
        }
        findPreference<Preference>("restore_internal_layout")?.setOnPreferenceClickListener {
            restoreInternalLayoutLauncher.launch(null)
            true
        }
        findPreference<Preference>("restore_external_layout")?.setOnPreferenceClickListener {
            restoreExternalLayoutLauncher.launch(null)
            true
        }
    }

    private fun setMirrorEnabled(mirrorPreference: SwitchPreference) {
        updatingMirrorPreference = true
        mirrorPreference.isChecked = true
        preferenceManager.sharedPreferences?.edit()?.putBoolean("save_internal_config_as_file", true)?.apply()
        updatingMirrorPreference = false
    }
}
